tests/migration/postcopy.go

```diff
@@ -0,0 +1,399 @@
+				})
+
+			})
+
+			Context("should migrate using for postcopy", func() {
+
+				applyMigrationPolicy := func(vmi *v1.VirtualMachineInstance) {
+					AlignPolicyAndVmi(vmi, migrationPolicy)
+					migrationPolicy = CreateMigrationPolicy(virtClient, migrationPolicy)
+				}
+
+				applyKubevirtCR := func() {
+					config := getCurrentKvConfig(virtClient)
+					config.MigrationConfiguration.AllowPostCopy = migrationPolicy.Spec.AllowPostCopy
+					config.MigrationConfiguration.CompletionTimeoutPerGiB = migrationPolicy.Spec.CompletionTimeoutPerGiB
+					config.MigrationConfiguration.BandwidthPerMigration = migrationPolicy.Spec.BandwidthPerMigration
+					tests.UpdateKubeVirtConfigValueAndWait(config)
+				}
+
+				type applySettingsType string
+				const (
+					applyWithMigrationPolicy applySettingsType = "policy"
+					applyWithKubevirtCR      applySettingsType = "kubevirt"
+				)
+
+				DescribeTable("[test_id:4747] using", func(settingsType applySettingsType) {
+					vmi := libvmi.NewFedora(libnet.WithMasqueradeNetworking()...)
+					vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("512Mi")
+					vmi.Spec.Domain.Devices.Rng = &v1.Rng{}
+					vmi.Namespace = testsuite.NamespacePrivileged
+
+					switch settingsType {
+					case applyWithMigrationPolicy:
+						applyMigrationPolicy(vmi)
+					case applyWithKubevirtCR:
+						applyKubevirtCR()
+					}
+
+					By("Starting the VirtualMachineInstance")
+					vmi = tests.RunVMIAndExpectLaunch(vmi, 240)
+
+					By("Checking that the VirtualMachineInstance console has expected output")
+					Expect(console.LoginToFedora(vmi)).To(Succeed())
+
+					// Need to wait for cloud init to finish and start the agent inside the vmi.
+					Eventually(matcher.ThisVMI(vmi), 12*time.Minute, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
+
+					runStressTest(vmi, "350M", stressDefaultSleepDuration)
+
+					// execute a migration, wait for finalized state
+					By("Starting the Migration")
+					migration := libmigration.New(vmi.Name, vmi.Namespace)
+					migration = libmigration.RunMigrationAndExpectToComplete(virtClient, migration, 150)
+
+					// check VMI, confirm migration state
+					libmigration.ConfirmVMIPostMigration(virtClient, vmi, migration)
+					confirmMigrationMode(virtClient, vmi, v1.MigrationPostCopy)
+				},
+					Entry("a migration policy", applyWithMigrationPolicy),
+					Entry("[Serial] Kubevirt CR", Serial, applyWithKubevirtCR),
+				)
+
+				Context("[Serial] and fail", Serial, func() {
+					var createdPods []string
+					BeforeEach(func() {
+						createdPods = []string{}
+					})
+
+					createLargeVirtualMachine := func(namespace string) *v1.VirtualMachine {
+						vmi := libvmi.NewFedora(libnet.WithMasqueradeNetworking()...)
+						vmi.Namespace = namespace
+						vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("3Gi")
+						vmi.Spec.Domain.Devices.Rng = &v1.Rng{}
```

Comment:
Probably we could save some time using Alpine or Cirros instead of Fedora. Here we don't need `stress-ng` :)
And in general we can drop the namespace parameter since we are creating the VM using directly:
`virtClient.VirtualMachine(testsuite.NamespacePrivileged)`.
Last, we can also use use libvmi Options:
```suggestion
						vmi := libvmi.NewFedora(
							append(libnet.WithMasqueradeNetworking(),
								libvmi.WithResourceMemory("2Gi"),
								libvmi.WithRng())...,
						)
```

Reply:
@fossedihelm, we need stress-ng here. 
1. to make sure there is enough time before the migration switches to post copy.
2. that the memory of that guest is already "polluted" so it will take time copy the rest of the memory to target after switch to post copy.

Reply:
Right! Sorry, I missed the line! Sorry for the noise